src/python_extension.ts

```diff
@@ -23,8 +24,27 @@ export class PythonExtension {
 		this.executionDetails = pythonApi.settings.getExecutionDetails(workspace.workspaceFolders?.[0].uri);
 		this.pythonApi = pythonApi;
 		this.onChange(pythonApi);
+
+	}
+
+	async hidePythonButtonCfg(): Promise<void> {
```

Comment:
I guess this works but I think it would prefer that our extension only fiddles with our own stuff and configs: So hide our button whenever the python one is visible. Did we had the case where the 2 where shown when a .cf file was open our only with .py files?

Reply:
> it would prefer that our extension only fiddles with our own stuff and configs

I agree, I will try something cleaner, and iirc it was only when opening .py files

Reply:
If I understand correctly this may invalidate the trouble you were still having with restoring the Python button so I'll hold off on looking into that for now.